core/txpool/legacypool/legacypool.go

```diff
@@ -1739,9 +1742,26 @@ func (pool *LegacyPool) demoteUnexecutables() {
 			// Internal shuffle shouldn't touch the lookup set.
 			pool.enqueueTx(hash, tx, false, false)
 		}
-		pendingGauge.Dec(int64(len(olds) + len(drops) + len(invalids)))
+
+		conditionalDrops, err := list.FilterTransactionConditionals(pool.currentState)
```

Comment:
This runs on every new block? That could be pretty expensive in the worst case especially with 2s blocktime. Not sure what the best semantics are here to be honest, it could be to run this every 10 blocks or something like that? Curious what you think @protolambda

Reply:
Every 10 blocks would hardcode the block time.

I think the ideal semantics, if it were an external thing, to look at the block witness diff, and invalidate txs quickly that way, without having to check every candidate tx storage slot. But inverting the diff like that, and accessing witness data, is going to be expensive and large diff. 

Also, doing everything only every N blocks results in lag spikes. Ideally we spread out the work.

Maybe just randomize the filter check, to sample the pool contents, to reduce workload without making it too manipulable or spiky?

Reply:
So I ended up just removing this filter. My rationale here is that we don't the miner will ultimately already check this when building the block and drop failed conditionals. I actually dont see the value add in trying to filter this again prior

what do yall think about this?

Reply:
Are we sure that when the miner drops the txs, they leave the mempool? Does the miner operate with copies of the txs?

Reply:
Ahh good catch! the miner operates on a copy. Didn't realize `demoteUnexecutables` was also a method of removing from the invalid txs from the mempool even when the same node has built the block. My initial understanding was that if it received a new block via p2p, this was a means of evicting bad txs.

will place it back in!

Reply:
The main scenario to handle here is when the miner drops a tx due to a failed conditional. If the conditional passes, the transaction will depart the mempool like any other transaction in `demoteUnexecutable`.

Instead of rechecking the conditional to detect transactions that enter a failed state from mempool -> miner, we could instead set a boolean reference on the conditional marking it as rejected. Then the mempool only needs to load this reference and not recheck the whole thing.

Will be good to track the metrics to see what % of txs race to be rejected by the miner. I'd anticipate that most invalidated conditional would fail at the submission stage

Reply:
@tynes see the latest changes that does the above. `LazyTransaction` fetches transactions from the pool by reference so we're good to directly set the `Rejected` field on the conditional

Reply:
Added a unit test for this as well in `worker_test`. Cheap to do every block for txs that have a conditional attached to them